src/instructlab/configuration.py

```diff
@@ -18,55 +18,139 @@
+    API_KEY = "no_api_key"
+
+    # TODO: Consolidate --model and --model-path into one --model-path flag since we always need a path now
+    MODEL_NAME = "merlinite-7b-lab-Q4_K_M"
+    MERLINITE_GGUF_REPO = "instructlab/merlinite-7b-lab-GGUF"
+    GGUF_MODEL_NAME = "merlinite-7b-lab-Q4_K_M.gguf"
+    MODEL_REPO = "instructlab/granite-7b-lab"
+    JUDGE_MODEL_MT = "prometheus-eval/prometheus-8x7b-v2.0"
+    TAXONOMY_REPO = "https://github.com/instructlab/taxonomy.git"
+    TAXONOMY_BASE = "origin/main"
+    MAX_CONTEXT_SIZE = 4096
+    # TODO: these constants should be removed, they should not leak out
+    NUM_CPUS = 10
+    CHUNK_WORD_COUNT = 1000
+    NUM_INSTRUCTIONS = 100
+    CONNECTION_TIMEOUT = httpx.Timeout(timeout=30.0)
+    # use spawn start method, fork is not thread-safe
+    MULTIPROCESSING_START_METHOD = "spawn"
+
+    # When otherwise unknown, ilab uses this as the default family
+    MODEL_FAMILY = "merlinite"
+
+    def __init__(self):
+        self._reset()
+
+    def _reset(self):
+        """
+        Utility function which is mostly used for testing purposes to clear the cache.
+        Otherwise, all tests will used cached temporary directories and cause errors.
+        """
+        pd = platformdirs.PlatformDirs(appname=ILAB_PACKAGE_NAME)
+        self._cache_home = pd.user_cache_dir
+        self._config_dir = pd.user_config_dir
+        self._data_dir = pd.user_data_dir
+
+    @property
+    def CHECKPOINTS(self) -> str:
+        return path.join(self._cache_home, STORAGE_DIR_NAMES.CHECKPOINTS)
+
+    @property
+    def DATASETS(self) -> str:
+        return path.join(self._data_dir, STORAGE_DIR_NAMES.DATASETS)
+
+    @property
+    def CONFIG_FILE(self) -> str:
+        return path.join(self._config_dir, CONFIG_FILENAME)
+
+    @property
+    def MODELS(self) -> str:
+        return path.join(self._data_dir, STORAGE_DIR_NAMES.MODELS)
+
+    @property
+    def DEFAULT_MODEL(self) -> str:
+        return path.join(self.MODELS, self.MODEL_NAME)
+
+    @property
+    def DEFAULT_GGUF_MODEL(self) -> str:
+        return path.join(self.MODELS, self.GGUF_MODEL_NAME)
+
+    @property
+    def TAXONOMY(self) -> str:
+        return path.join(self._data_dir, STORAGE_DIR_NAMES.TAXONOMY)
+
+    @property
+    def CHATLOGS(self) -> str:
+        return path.join(self._data_dir, STORAGE_DIR_NAMES.CHATLOGS)
+
+    @property
+    def INTERNAL(self) -> str:
+        return path.join(self._data_dir, STORAGE_DIR_NAMES.INTERNAL)
+
+    @property
+    def PROMPT_FILE(self) -> str:
+        return path.join(self.INTERNAL, "prompt.txt")
+
+    @property
+    def SEED_FILE(self) -> str:
+        return path.join(self.INTERNAL, "seed_tasks.json")
+
+    @property
+    def EVAL_DATA(self) -> str:
+        return path.join(self.INTERNAL, "eval_data")
+
+
+DEFAULTS = _InstructlabDefaults()
+
 
 # Model families understood by ilab
-MODEL_FAMILIES = set(("merlinite", "mixtral"))
+MODEL_FAMILIES = {"merlinite", "mixtral"}
```

Comment:
granite | mistral | dolomite?

Reply:
It looks like there's some mapping logic which exists to handle the Granite case: https://github.com/instructlab/instructlab/blob/a94048fd65ddaa5e2a38b8034536a584897af832/src/instructlab/configuration.py#L63 

We can open an issue for this if you'd like, but we'd need to handle this in a separate PR.

Reply:
++